calc/vxlan_resolver.go

```diff
@@ -263,6 +265,48 @@ func (c *VXLANResolver) OnHostConfigUpdate(update api.Update) (_ bool) {
 			})
 			c.sendVTEPRemove(nodeName)
 		}
+	case "VXLANTunnelMACAddr":
+		nodeName := update.Key.(model.HostConfigKey).Hostname
+		vtepSent := c.vtepSent(nodeName)
+		logCxt := logrus.WithField("node", nodeName).WithField("value", update.Value)
+		logCxt.Debug("VXLANTunnelMACAddr update")
+		if update.Value != nil {
+			// Update for a VXLAN tunnel MAC address.
+			newMAC := update.Value.(string)
+			currMAC := c.nodeNameToVXLANMac[nodeName]
+			logCxt = logCxt.WithFields(logrus.Fields{"newMAC": newMAC, "currMAC": currMAC})
+			if vtepSent {
+				if currMAC == newMAC {
+					// If we've already handled this node, there's nothing to do. Deduplicate.
+					logCxt.Debug("Skipping duplicate tunnel MAC addr update")
+					return
+				}
+
+				// Try sending a VTEP update. If we do, this will trigger a kick of any
+				// pending routes which might now be ready to send.
+				c.nodeNameToVXLANMac[nodeName] = newMAC
+				if c.sendVTEPUpdate(nodeName) {
+					// We've successfully sent a new VTEP - check to see if any pending routes
+					// are now ready to be programmed.
+					logCxt.Info("Sent VTEP to dataplane, check pending routes")
+					c.kickPendingRoutes(pendingSet)
+				}
+			}
+
+		} else {
+			logCxt.Info("Update the VTEP with the system generated MAC address and send it to dataplane")
+			_, ok := c.nodeNameToVXLANMac[nodeName]
+			if ok {
+				delete(c.nodeNameToVXLANMac, nodeName)
+			}
+
+			if c.sendVTEPUpdate(nodeName) {
+				// We've successfully sent a new VTEP - check to see if any pending routes
+				// are now ready to be programmed.
+				logCxt.Info("Sent VTEP to dataplane, check pending routes")
+				c.kickPendingRoutes(pendingSet)
```

Comment:
I don't think we can actually kick any routes here since adding/removing the MAC can't make the VTEP valid/invalid. Could probably remove this call to kickPendingRoutes()

Reply:
Acknowledged and changed the code.